### ISSUE TITLE
Scala syntax highlighting on GitHub.com looks broken

### ISSUE THREAD

**Issue by olafurpg** [scala/vscode-scala-syntax#33]
The `val` and `def` keywords are colored as black

<img width="612" alt="Screenshot 2019-06-06 at 13 34 44" src="https://user-images.githubusercontent.com/1408093/59031808-02b4f980-8865-11e9-8f5d-e746f8a166b9.png">

There was a new release of github/linguist this morning which is currently live: https://github.com/github/linguist/pull/4547

The most likely cause for this regression is the change from `Scala.tmLanguage` to `Scala.tmLanguage.json` in https://github.com/scala/vscode-scala-syntax/pull/20, cc/ @PanAeon

**Comment by olafurpg** [comment]
The diff for Scala syntax highlighting definition in linguist between v7.4 (latest correct release) and v7.5 (current broken release) https://gist.github.com/olafurpg/125fc86cfb4d6a3728285411a5bdbcc2

**Comment by olafurpg** [comment]
For detailed comparison, here are the two copies of `source.scala.json` for linguist v7.4 and v7.5

[scala.zip](https://github.com/scala/vscode-scala-syntax/files/3261964/scala.zip)

**Comment by olafurpg** [comment]
To test GitHub highlighting given a grammar it's possible to use Lightshow

- https://github-lightshow.herokuapp.com/?utf8=%E2%9C%93&scope=from-url&grammar_format=auto&grammar_url=https%3A%2F%2Fgist.githubusercontent.com%2Folafurpg%2F848cadc0161be1c008d6fd77c6840c6d%2Fraw%2F08077b41d59ca0de1172a77ca123618614b0297e%2Fscala-7.4.json&grammar_text=&code_source=from-text&code_url=&code=val+x+%3D+42
- https://github-lightshow.herokuapp.com/?utf8=%E2%9C%93&scope=from-url&grammar_format=auto&grammar_url=https%3A%2F%2Fgist.githubusercontent.com%2Folafurpg%2Ff7dca526a5b290f88082804ed0800836%2Fraw%2F897c80f510f6e2dd2202b4dd5729566b57814ed6%2Fscala-7.5.json&grammar_text=&code_source=from-text&code_url=&code=val+x+%3D+42

**Comment by PanAeon** [comment]
@olafurpg Hi, sorry for the trouble, the fact that github uses PCRE is a total surprise to me. Since linguist's contributing guide refer to the [textmate's documentation](https://manual.macromates.com/en/language_grammars) I've assumed it also uses oniguruma. :'(

I've tested the new `val` and `def` regex rules on regex101.com, and they are indeed broken. So far I have found the  problem with character class union which is not supported by PCRE, e.g. `[..[...]..[...]]`, and also character class subtraction. On the positive side, it looks like all other advanced features used in the grammar are supported, but I never worked with PCRE and will have to check its documentation to be sure.

I'll try to investigate this in more detail tomorrow, but I'll be on vacation from Saturday 08th until Tuesday 18th, and it's unlikely that I'll  be able to help with the issue during this time ...
